nixos/lib/make-options-doc/default.nix

```diff
@@ -1,20 +1,106 @@
-/* Generate JSON, XML and DocBook documentation for given NixOS options.
+/**
+  Generates documentation for nix modules.
 
-   Minimal example:
+  # Outputs
 
-    { pkgs,  }:
+  This function offers the following outputs
 
-    let
-      eval = import (pkgs.path + "/nixos/lib/eval-config.nix") {
-        baseModules = [
-          ../module.nix
-        ];
-        modules = [];
-      };
-    in pkgs.nixosOptionsDoc {
-      options = eval.options;
+  ## optionsCommonMark
+
+  Documentation in CommonMark text format.
+
+  ## optionsJSON
+
+  All options in a JSON format suitable for further automated processing.
+
+  `example.json`
+  ```json
+  {
+    ...
+    "fileSystems.<name>.options": {
+      "declarations": ["nixos/modules/tasks/filesystems.nix"],
+      "default": {
+        "_type": "literalExpression",
+        "text": "[\n  \"defaults\"\n]"
+      },
+      "description": "Options used to mount the file system.",
+      "example": {
+        "_type": "literalExpression",
+        "text": "[\n  \"data=journal\"\n]"
+      },
+      "loc": ["fileSystems", "<name>", "options"],
+      "readOnly": false,
+      "type": "non-empty (list of string (with check: non-empty))"
+    },
+    ...
+  }
+  ```
+
+  Overall the following fields are available per option:
+
+  ```
+  declarations
+  description
+  loc
+  readOnly
+  type
+  default
+  example
+  relatedPackages
+  ```
+
+  ## optionsDocBook
+
+  deprecated since 23.11 and will be removed in 24.05.
+
+  ## optionsAsciiDoc
+
+  Documentation rendered as AsciiDoc. This is useful for e.g. man pages.
+
+  > Note: NixOS uses this to build the man-pages available via `man configuration.nix` on every NixOS machine.
```

Comment:
Nit: "man pages", "every NixOS machine" is still wrong. Maybe just "to build the `configuration.nix` manual page"?

Reply:
Ok. I still would like to use "man page" because its a technical term. manual page could also be confused with the web page.

Reply:
Yeah, that's probably better.